python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -167,7 +168,19 @@ def _update(self):
         horizon = now - (60 * self.config["idle_timeout_minutes"])
 
         nodes_to_terminate = []
+        node_type_counts = collections.defaultdict(int)
         for node_id in nodes:
+            # Make sure to not kill idle node types if the number of workers
+            # of that type is lower/equal to the min_workers of that type.
```

Comment:
Can you pull this out into a helper function to avoid making this method too long?